### DISCUSSION TITLE
Python

### DISCUSSION THREAD

**Discussion by simon14789** [community/community#21418] [Other Feature Feedback, Questions, & Ideas]
<h1>it’s so funny</h1>
<p># if you can’t understand,please run it in your IDE</p>
<h1>by the way,i’m chinese</h1>
<p>letter = [‘a’,‘b’,‘c’,‘d’,‘e’,‘f’,‘g’,‘h’,‘i’,‘j’,‘k’,‘l’,‘m’,‘n’,‘o’,‘p’,‘q’,‘r’,‘s’,‘t’,‘u’,‘v’,‘w’,‘x’,‘y’,‘z’,’ ']<br>
number = [8,26,13,4,4,3,26,0,26,5,14,17,4,8,6,13,26,5,17,8,4,13,3]<br>
message = “”</p>
<p>for i in number:<br>
message += letter[i]</p>

**Comment by MChevy422** [comment]
<p>Hello <a class="mention" href="/simon14789">@simon14789</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the Programming Help and Discussion board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Greatly appreciated,</p>
<p><strong>-MChevy422</strong></p>

**Reply by HenryVince** [reply]
Hello @simon14789,

Thank you for the update and for moving my post to the Programming Help and Discussion board. I appreciate your effort in keeping the community organized.

I do have another question related to my website (https://notoxmd.com/). Recently, I've been encountering a specific issue with the website's performance. Users have reported slow loading times, and sometimes the site even times out. I've checked the server logs, and there don't seem to be any significant errors.

I've tried optimizing the code, compressing images, and implementing caching mechanisms, but the problem persists. I'm reaching out to the community for suggestions or insights into what might be causing these performance issues and how I can address them.

If anyone has experience dealing with similar problems or has suggestions on tools to diagnose and optimize website performance, I would greatly appreciate your input.

Thank you in advance for any assistance.

**Comment by Wabri** [comment, marked as answer]
<p>The code write like this cannot work, the for loop need spaces. Something like this:</p>
<pre><code>letter = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z',' ']
number = [8,26,13,4,4,3,26,0,26,5,14,17,4,8,6,13,26,5,17,8,4,13,3]
message = ""

for i in number:
    message += letter[i]
</code></pre>
<p>Another thing you need to print the output otherwise it will not print anything:</p>
<pre><code>letter = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', ' ']
number = [8, 26, 13, 4, 4, 3, 26, 0, 26, 5, 14, 17, 4, 8, 6, 13, 26, 5, 17, 8, 4, 13, 3]
message = ""

for i in number:
    message += letter[i]

print(message)
</code></pre>
<p>One last thing, please read the “Conventions for the Python code” here: <a href="https://www.python.org/dev/peps/pep-0008/" rel="nofollow noopener">https://www.python.org/dev/peps/pep-0008/</a></p>
<p>The code will be more readable and beauty if you follow it:</p>
<pre><code>letter = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
          'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', ' ']
number = [8, 26, 13, 4, 4, 3, 26, 0, 26, 5, 14,
          17, 4, 8, 6, 13, 26, 5, 17, 8, 4, 13, 3]
message = ""

for i in number:
    message += letter[i]

print(message)
</code></pre>
<p>If you do that we can be friends :slightly_smiling_face:</p>
<p>-Gabriele-</p>

**Reply by rizem441** [reply]
Ok

**Comment by simon14789** [comment]
<p>What!Where is my space!There are spaces in my IDE,But I don’t know why it’s gone when it’s copied.Thank you for helping me find it.I’m going to reissue it now.What text editor do you use? It looks good. It’s different from mine.</p>

**Comment by Wabri** [comment]
<p>I use Atom for scripting languages</p>

**Comment by simon14789** [comment]
<p>How long do programmers work in your country?I’m working now. Most Chinese programmers should be working now!Working hours in China range from 8 a.m. to around 10 p.m.</p>

**Comment by ThirdEyeRose** [comment]
<p>Good question <a class="mention" href="/simon14789">@simon14789</a>! I can only speak for my country and I live in the United States.</p>
<p>Here, programmers tend to have a decent amount of freedom to choose a schedule that works for them as they can work for big corporations, small startups, or even as independent contractors.</p>
<p>As a cultural average, people tend to try to work 35-45 hours a week. However, it’s not uncommon for some companies to expect their workers to work 60 hours during certain weeks when they’re trying to get things done before a deadline. These are called crunch weeks. There are mixed feelings about crunch periods as a lot can get done, but they tend to produce sloppy work and cause burnout for people as they’re pushed beyond their limit of being able to take care of themselves.</p>

**Comment by marcsultan** [comment]
<p> I am so new to Python, I can hardly understand anything</p>
<p>I am taking it this semester, can anyone volunteer to help me please</p>

**Comment by fire-eggs** [comment]
<blockquote>
<p><a class="mention" href="/marcsultan">@marcsultan</a> wrote:</p>
</blockquote>
<blockquote>
<p> I am so new to Python, I can hardly understand anything</p>
</blockquote>
<blockquote>
<p>I am taking it this semester, can anyone volunteer to help me please</p>
</blockquote>
<p>Feel free to post your questions or confusion as new topics in this forum.</p>
<p>Good luck with your studies!</p>

**Comment by MoejoMan** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/4/d43838d632439a6ca0322211c065e20a1be78ea3.png" data-download-href="/uploads/short-url/uhnG5vVmPcy7K37ZMeuzQ3vAfuP.png?dl=1" title="image_2022-02-01_231400" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/68347577/181064495-6da822e7-7188-4cf2-875d-93df71f242c6.png" alt="image_2022-02-01_231400" data-base62-sha1="uhnG5vVmPcy7K37ZMeuzQ3vAfuP" width="690" height="330" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d43838d632439a6ca0322211c065e20a1be78ea3_2_690x330.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d43838d632439a6ca0322211c065e20a1be78ea3_2_1035x495.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d43838d632439a6ca0322211c065e20a1be78ea3_2_1380x660.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d43838d632439a6ca0322211c065e20a1be78ea3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image_2022-02-01_231400</span><span class="informations">1903×912 58 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>i have no clue why this doesn’t work, its a random number generator from 1-100, which generates 50 numbers, then asks for user input to see if that number is contained in the list. any help?</p>

**Comment by airtower-luna** [comment]
<p>You have an <code>else</code> without <code>if</code>. An <code>else</code> block must follow immediately after the matching <code>if</code> (or <code>elif</code>) block.</p>
<p>In Python indentation determines blocks, so the <code>if</code> block that starts on line 13 ends with the unindented line 15. If you want lines 15 and 16 to be part of the block you need to indent them just like line 14.</p>
